Source/JavaScriptCore/llint/InPlaceInterpreter.asm

```diff
@@ -333,7 +338,39 @@ macro operationCallMayThrow(fn)
     fn()
     bpneq r1, (constexpr JSC::IPInt::SlowPathExceptionTag), .continuation
     storei r0, ArgumentCountIncludingThis + PayloadOffset[cfr]
+    addq 4 * MachineRegisterSize, sp
+    jmp _wasm_throw_from_slow_path_trampoline
+.continuation:
+    if ARM64 or ARM64E
+        pop ws0, PL
+    elsif X86_64
+        addq MachineRegisterSize, sp
+        pop PL
+    end
+    pop MC, PC
+end
+
+macro operationCallMayThrowPreservingRegisters(fn)
+    // FIXME: preserveWasmArgumentRegisters() and restoreWasmArgumentRegisters() are not safe for SIMD.
+    preserveWasmArgumentRegisters()
+    saveCallSiteIndex()
+
+    move wasmInstance, a0
+    push PC, MC
+    if ARM64 or ARM64E
+        push PL, ws0
+    elsif X86_64
+        push PL
+        # preserve 16 byte alignment.
+        subq MachineRegisterSize, sp
+    end
+    fn()
+    bpneq r1, (constexpr JSC::IPInt::SlowPathExceptionTag), .continuation
+
+    storei r0, ArgumentCountIncludingThis + PayloadOffset[cfr]
+    addq (NumberOfWasmArgumentGPRs + NumberOfWasmArgumentFPRs + 4) * 8, sp
```

Comment:
Should this be `addq (NumberOfWasmArgumentGPRs + NumberOfWasmArgumentFPRs) * 8 + 4 * MachineRegisterSize, sp`?

Reply:
It should be `addp (NumberOfWasmArgumentGPRs + NumberOfWasmArgumentFPRs + 4) * MachineRegisterSize, sp`.  Wasm GPRs and FPRs are also of size `MachineRegisterSize`.  This works until IPInt implements SIMD support (see detail (8) in the commit message).

Reply:
`preserveWasmGPRArgumentRegisters` shows gprs are in `MachineRegisterSize` and `preserveWasmArgumentRegisters` shows fprs are in 8-byte size. So, should this be `addp (NumberOfWasmArgumentGPRs + 4) * MachineRegisterSize + NumberOfWasmArgumentFPRs * 8, sp
`?